piqe_ocp_lib/api/resources/ocp_health_checker.py

```diff
@@ -0,0 +1,318 @@
+            for node_info in node_list_info.items:
+                temp_list = list()
+                for condition in node_info.status.conditions:
+                    if condition["type"] == "Ready":
+                        individual_node_health_status_list.append(condition["status"])
+                    elif condition["type"] == "MemoryPressure" and condition["status"] == "True":
+                        temp_list.append({"MemoryPressure": "The node memory is low"})
+                    elif condition["type"] == "DiskPressure" and condition["status"] == "True":
+                        temp_list.append({"DiskPressure": "The disk capacity is low"})
+                    elif condition["type"] == "PIDPressure" and condition["status"] == "True":
+                        temp_list.append({"PIDPressure": "There are too many processes on the node"})
+
+                unhealthy_node_info[node_info["metadata"]["name"]] = temp_list
+
+        if len(set(individual_node_health_status_list)) == 1 and \
+                list(set(individual_node_health_status_list))[0] == "True":
+            all_nodes_healthy = True
+
+        return all_nodes_healthy, unhealthy_node_info
+
+    def check_router_health(self):
+        """
+        Check openshift router health
+        - Check if router pod is running fine in openshift-ingress namespace
+        - Check if deployments of router has matching number of replicas
+        :return:Return tuple of bool (is_router_healthy) and dict (unhealthy_router_info)
+        """
+        logger.info("Check the health of openshift router operator pod")
+        is_router_healthy = False
+        unhealthy_router_info = dict()
+        is_router_pod_healthy = False
+        is_replicas_count_matching = False
+        unhealthy_pods = dict()
+        pods_response = self.ocp_pod.list_pods_in_a_namespace(namespace="openshift-ingress")
+        for pod in pods_response.items:
+            for condition in pod["status"]["conditions"]:
+                if condition["type"] == "Ready" and condition["status"] == "False":
+                    unhealthy_pods[pod["metadata"]["name"]] = condition["status"]
+
+        if len(unhealthy_pods.values()) == 0:
+            is_router_pod_healthy = True
+
+        unhealthy_router_info.update({"router_pod": unhealthy_pods.keys()})
+        logger.info("Unhealthy router info : %s", unhealthy_router_info)
+        logger.info("Is router pod/s healthy : %s", is_router_pod_healthy)
+
+        logger.info("Check replicas count of openshift router deployment are matching")
+        deployment_response = self.ocp_deployment.list_all_deployments_in_a_namespace(namespace="openshift-ingress")
+        for deployment in deployment_response.items:
+            if deployment.metadata.name == "router-default":
+                replicas = deployment.status.replicas
+                available_replicas = deployment.status.availableReplicas
+                ready_replicas = deployment.status.readyReplicas
```

Comment:
There may be a refactoring opportunity between this and the code on line 154.

Reply:
Same as above